alf/algorithms/mcts_models.py

```diff
@@ -174,8 +192,18 @@ def recurrent_inference(self, state, action):
         Returns:
             ModelOutput: the prediction
         """
-        new_state = self._dynamics_net((state, action))[0]
-        return self.prediction_model(new_state)
+        if not self._handle_bn:
+            new_state = self._dynamics_net((state, action))[0]
+            return self.prediction_model(new_state)
+        else:
+            state, current_steps = state
+            self._prediction_net.set_batch_norm_current_step(current_steps)
+            new_state = self._dynamics_net((state, action))[0]
+            current_steps = current_steps + 1
+            self._dynamics_net.set_batch_norm_current_step(current_steps)
+            model_output = self.prediction_model(new_state)
+            return model_output._replace(
+                state=(model_output.state, current_steps))
```

Comment:
Such "tuple for RNN state where the first is the hidden and the 2nd is the steps" is implicitly used in a lot of places - maybe a good indicator to create a `NamedTuple` for this so that both fields are named?

Reply:
good suggestion